docs/proposals/multiple-sources-for-applications.md

```diff
@@ -0,0 +1,225 @@
+sponsors:
+  - TBD        # List all intereste parties here.
+reviewers:
+  - "@jannfis"
+  - TBD
+approvers:
+  - "@jannfis"
+  - TBD
+
+creation-date: 2022-01-28
+last-updated: 2022-01-28
+---
+
+# Multiple Sources for application
+
+Support more than one source for creating an Application.
+
+Related Issues: 
+* https://github.com/argoproj/argo-cd/issues/677
+* https://github.com/argoproj/argo-cd/issues/2789
+
+## Open Questions
+* Adding external sources to the Application reource would add additional latencies for creation/reconcilation process. Should we add it to the doc in Risks?
+* 
+
+## Summary
+
+Currently, Argo CD supports Applications with only a single ApplicationSource. In certain scenarios, it would be useful to support more than one source for the application. For example, consider a need to create multiple applications with the same manifests. Today we would have to copy manifest files from one application to another.
+
+Creating applications from multiple sources would allow users to configure multiple services stored at various sources within the same application.
+
+## Motivation
+
+The main motivation behind this enhancement proposal is to allow users to create an application using services that are stored in various sources.
+
+### Goals
+
+The goals of the enhancement are:
+
+#### **Supporting multiple sources for creating an application**
+
+Users should be able to specify multiple sources to add services to the application. Argo CD should compile all the sources and reconcile each source individually for creating the application.
+
+#### **Allow specifying external value files for Helm repositories**
+
+Users should be able to specify different sources for Helm charts and values files. The Helm charts specified by the user could be available in Git or Helm repository and the value files are stored in Git. Argo CD should track changes in both the Helm charts and the value files repository and reconcile the application.
+
+#### Changes to UI
+
+The UI should allow users to add multiple sources while creating the application. For each source, UI should allow users to add multiple external values files Helm projects.
+
+#### Changes to cli
+
+The cli would need to support adding a list of resources instead of just one while creating the application. Also, just like `--values` field for adding values files, we would need an option to allow external value files to the application.
+
+### Non-goals
+* Allow reconciliation from Argo CD Application resources that are located at various sources. We believe this would be possible with some adaptions in the reconcilation workflow.
+
+## Proposal
+
+### Add new `sources` field in Application Spec
+
+The proposal is to add a new field `sources` which would allow users to input list of `ApplicationSource`s. We would mark field `source` as deprecated and would override the details under `source` with details under `sources` field.
+
+```yaml
```

Comment:
Can you describe what this example is meant to communicate? I think it's supposed to show old-style vs. new-style sources, but it reads a bit like "here's an Application with the same source twice".

Reply:
I have updated the example to a real world example. It might give more details on how it should be. But the basic idea is, if both `source` and `sources` field would be provided, we will ignore the `source` field.